lib/sproxyd.js

```diff
@@ -287,79 +287,76 @@ class SproxydClient {
         const host = this.getCurrentBootstrap();
         const isBatchDelete = key === '.batch_delete';
         const newKey = key || keygen(this.cos, params);
-        const req = this._createRequestHeader(method, headers, newKey, params,
-            log);
+        const req = this._createRequestHeader(method, headers, newKey, params, log);
+        const logInfo = {
+            component: 'sproxydclient',
+            method: '_handleRequest',
+            host,
+            key: newKey,
+            contentLength: size,
+        };
 
         if (stream) {
-            let isProcessingPUT = false;
+            let streamingStarted = false;
             let voluntaryAbort = false;
             headers['content-length'] = size;
 
             const request = _createRequest(req, log, (err, response) => {
                 if (err) {
-                    if (isProcessingPUT || voluntaryAbort) {
+                    if (streamingStarted || voluntaryAbort) {
                         err.retryable = false;
                     } else {
                         err.retryable = true;
                     }
                     if (!voluntaryAbort) {
-                        log.error('putting chunk to sproxyd', { host, key: newKey, error: err });
+                        logInfo.error = err;
+                        log.error('putting chunk to sproxyd', logInfo);
```

Comment:
can we use spread instead?

```suggestion
                        log.error('putting chunk to sproxyd', { ...logInfo, error: err })
```

Reply:
Thanks! Won't be needed this time, but I'll keep the syntax in mind for later.